src/main/java/seedu/duke/util/Parser.java

```diff
@@ -234,6 +250,73 @@ private Command prepareAddNote(String userMessage) throws SystemException {
         }
     }
 
+    private String checkBlank(String input, SystemException.ExceptionType e) throws SystemException {
+        if (input.isBlank()) {
+            throw new SystemException(e);
+        } else {
+            return input.trim();
+        }
+    }
+
+    private Command prepareAddEvent(String userMessage) throws SystemException {
+        // add-e eventTitle /t timing /r occurrance /a time before (default same day)
+
+        String title = "";
+        LocalDateTime dateTime = null;
+        boolean toRemind = false;
+        boolean isRecurring = false;
+
+        try {
+            ArrayList<String[]> splitInfo = splitInfoDetails(userMessage);
+            for (String[] infoDetails : splitInfo) {
+                String prefix = infoDetails[0];
+                ExceptionType e;
+                switch (prefix) {
+                case PREFIX_TITLE:
+                    e = SystemException.ExceptionType.EXCEPTION_MISSING_TITLE;
+                    title = checkBlank(infoDetails[1], e);
+                    break;
+                case PREFIX_TIMING:
+                    e = SystemException.ExceptionType.EXCEPTION_MISSING_TIMING;
+                    String timingString = checkBlank(infoDetails[1], e);
+                    dateTime = dateTimeParser(timingString);
+                    break;
+                case PREFIX_REMIND:
+                    toRemind = true;
+                    break;
+                case PREFIX_RECURRING:
+                    isRecurring = true;
+                    break;
+                default:
+                    throw new SystemException(SystemException.ExceptionType.EXCEPTION_WRONG_PREFIX);
+                }
+            }
+
+            if (title.isBlank()) {
+                throw new SystemException(SystemException.ExceptionType.EXCEPTION_MISSING_TITLE);
+            } else if (dateTime == null) {
+                throw new SystemException(SystemException.ExceptionType.EXCEPTION_MISSING_TIMING);
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new SystemException(ExceptionType.EXCEPTION_MISSING_DESCRIPTION);
+        }
```

Comment:
There should be a catch for NullPointerException in the event that the user just enter the command only without any space after it. i.e. "add-e"

Reply:
Good catch! Didn't see this